framework/aster-frame/src/arch/x86/mm/mod.rs

```diff
@@ -239,3 +239,86 @@ impl fmt::Debug for PageTableEntry {
             .finish()
     }
 }
+
+/// Copies `count * size_of::<T>()` bytes from `src` to `dst`.
+/// The source and destination may overlap.
+///
+/// If the source and destination will never overlap, `fast_copy_nonoverlapping` can be used instead.
+///
+/// # Performance
+///
+/// This function is provided as a fast alternative to `core::ptr::copy`
+/// by selectively using the optimised `fast_copy_nonoverlapping` when it detects that
+/// the source and destination do not overlap.
+/// When overlap is detected, it falls back to the standard `core::ptr::copy`,
+/// which is safe for overlapping memory regions.
+///
+/// # Safety
+///
+/// The safety requirements of this function are consistent with `core::ptr::copy`.
+#[inline]
+pub unsafe fn fast_copy<T>(src: *const T, dst: *mut T, count: usize) {
+    if src == dst || count == 0 {
+        return;
+    }
+
+    let is_overlapping = {
+        let src_end = src.add(count);
+        let dst_end = dst.add(count);
+        src < dst_end && dst < src_end as *mut T
+    };
+
+    if !is_overlapping {
+        fast_copy_nonoverlapping(src, dst, count);
+    } else {
+        core::ptr::copy(src, dst, count);
```

Comment:
The overlapped case can also be handled gracefully with `movsb`-family instructions. So I think we can write assembly and don't need to fallback to `ptr::copy`.